client/chat/channel-status-card.tsx

```diff
@@ -0,0 +1,200 @@
+  const isUserBanned =
+    isFetchError(joinChannelError) && joinChannelError.code === ChatServiceErrorCode.UserBanned
+
+  let subtitle
+  if (isChannelNotFound) {
+    subtitle = <ErrorText>We couldn't find this channel, you may create it yourself</ErrorText>
+  } else if (isChannelClosed) {
+    subtitle = <ErrorText>We couldn't find this channel, it might have been closed</ErrorText>
+  } else if (channelStatus?.private && !isUserInChannel) {
+    subtitle = <ErrorText>This channel is private and requires an invite to join</ErrorText>
+  } else if (isUserBanned) {
+    subtitle = <ErrorText>You are banned from this channel</ErrorText>
+  } else if (channelStatus?.userCount) {
+    subtitle = (
+      <Body1>{`${channelStatus.userCount} member${channelStatus.userCount > 1 ? 's' : ''}`}</Body1>
+    )
+  }
+
+  let action
+  if (isUserInChannel) {
+    action = <RaisedButton label='View' onClick={onViewClick} />
+  } else if (isChannelNotFound) {
+    action = <RaisedButton label='Create' onClick={onJoinClick} />
+  } else if (channelStatus?.private || isUserBanned) {
+    action = <RaisedButton label='Join' disabled={true} />
+  } else if (!findChannelError) {
+    action = <RaisedButton label='Join' onClick={onJoinClick} />
+  }
+
+  return (
+    <Container>
+      {!channelStatus && !findChannelError ? (
+        <StyledLoadingDotsArea />
+      ) : (
+        <>
+          {icon}
+          <ChannelInfoContainer>
+            <Headline6>{channelName}</Headline6>
+            {subtitle}
+          </ChannelInfoContainer>
+          {action}
+        </>
+      )}
+    </Container>
+  )
+}
+
+export interface ConnectedChannelStatusCardProps {
+  channelId: SbChannelId
+  channelName: string
+}
+
+export function ConnectedChannelStatusCard({
+  channelId,
+  channelName,
+}: ConnectedChannelStatusCardProps) {
+  const dispatch = useAppDispatch()
+  const channel = useAppSelector(s => s.chat.byId.get(channelId))
+  const channelStatus = useAppSelector(s => s.channelStatus.byId.get(channelId))
+
+  const [findChannelError, setFindChannelError] = useState<Error>()
+  const [joinChannelError, setJoinChannelError] = useState<Error>()
+  const cancelJoinRef = useRef(new AbortController())
+
+  useEffect(() => {
+    const abortController = new AbortController()
+
+    dispatch(
+      findChannel(channelId, channelName, {
+        signal: abortController.signal,
+        onSuccess: () => setFindChannelError(undefined),
+        onError: err => setFindChannelError(err),
+      }),
+    )
+
+    return () => {
+      abortController.abort()
+    }
+  }, [channelId, dispatch, channelName])
+
+  useEffect(() => {
+    const abortController = cancelJoinRef.current
+
+    return () => {
+      abortController.abort()
+    }
+  }, [])
+
+  const onViewClick = useStableCallback(() => {
+    navigateToChannel(channelId, channelName)
+  })
+
+  const onJoinClick = useStableCallback(() => {
+    dispatch(
```

Comment:
This should maybe set some kind of flag to indicate a join is in progress an disable whatever button